bundles/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchConnector.java

```diff
@@ -0,0 +1,387 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ecotouch.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.util.*;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Network communication with Waterkotte EcoTouch heat pumps.
+ *
+ * The communication protocol was reverse engineered from the Easy-Con Android
+ * app. The meaning of the EcoTouch tags was provided by Waterkotte's technical
+ * service (by an excerpt of a developer manual).
+ *
+ * @author Sebastian Held <sebastian.held@gmx.de> - Initial contribution
+ * @since 1.5.0
+ */
+
+public class EcoTouchConnector {
+    private String ip;
+    private String username;
+    private String password;
+    List<String> cookies;
+    static Pattern responsePattern = Pattern.compile("#(.+)\\s+S_OK[^0-9-]+([0-9-]+)\\s+([0-9-.]+)");
+
+    private final Logger logger = LoggerFactory.getLogger(EcoTouchConnector.class);
+
+    /**
+     * Create a network communication without having a current access token.
+     */
+    public EcoTouchConnector(String ip, String username, String password) {
+        this.ip = ip;
+        this.username = username;
+        this.password = password;
+        this.cookies = null;
+    }
+
+    /**
+     * Create a network communication with access token. This speeds up
+     * retrieving values, because the log in step is omitted.
+     */
+    public EcoTouchConnector(String ip, String username, String password, List<String> cookies) {
+        this.ip = ip;
+        this.username = username;
+        this.password = password;
+        this.cookies = cookies;
+    }
+
+    private synchronized void trylogin(boolean force) throws Exception {
+        logger.debug("trylogin: in");
+        if (force == false && cookies != null) {
+            logger.debug("trylogin: out fast");
+            return;
+        }
+        login();
+        logger.debug("trylogin: out normal");
+    }
+
+    private void login() throws Exception {
+        logger.debug("login");
```

Comment:
Please read the guidelines on logging here
https://www.openhab.org/docs/developer/guidelines.html#f-logging
Logs should not replace the debugger. Can it be assumed that login passes if the thing comes ONLINE? Is the line useful or does it just increase the load on the system?

Reply:
I've reduced logging